File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.java
class Solution {
    public int maxCoins(int[] piles) {
        Arrays.sort(piles);
        
        int i = 0;
        int j = piles.length - 2;
        
        int total = 0;
        
        while(i < j){
            total += piles[j];
            i++;
            j -= 2;
        }
        
        return total;
    }
}

File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.py
class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        piles.sort()
        triplets = []
        l, r = 0, len(piles) - 1
        while l < r:
            triplet = [piles[r], piles[r - 1], piles[l]]
            triplets.append(triplet)
            r -= 2
            l += 1
        count = 0

        for p in triplets:
            count += p[1]
        return count